packages/instantsearch-react/src/connectors/connectSearchBox.js

```diff
@@ -37,9 +37,16 @@ export default createConnector({
     return searchParameters.setQuery(getQuery(props, state));
   },
 
-  getMetadata(props) {
+  getMetadata(props, state) {
+    const query = getQuery(props, state);
     return {
       id: props.id,
+      filters: !query ? [] : [{
+        key: props.id,
+        label: query,
+        hide: true,
```

Comment:
We have two different widgets that consume the metadata declared by other widgets: `Reset` and `CurrentFilters`.

* `CurrentFilters` displays filters from `RefinementList`, `Menu`, `Toggle`, etc. However, it doesn't display the current query nor the current page. And when you click on its `Clear All` button, it doesn't clear them either.
* `Reset` clears *everything*.

As such, I've made a distinction between two types of filters: visible filters (that are displayed in `CurrentFilters`) and invisible filters, that aren't displayed nor cleared by `CurrentFilters` but that are correctly cleared by `Reset`. Invisible filters are declared by setting the `hide` property to `true` on a filter object.